src/driver/op.rs

```diff
@@ -202,11 +254,11 @@ impl<T, CqeType> Drop for Op<T, CqeType> {
                     }
                 };
 
-                match mem::replace(lifecycle, Lifecycle::Submitted) {
+                match mem::replace(lifecycle, Lifecycle::Ignored(Box::new(()))) {
                     Lifecycle::Submitted | Lifecycle::Waiting(_) => {
                         *lifecycle = Lifecycle::Ignored(Box::new(self.data.take()));
                     }
-                    Lifecycle::Completed(..) => {
+                    Lifecycle::Completed(..) | Lifecycle::Pending(_) => {
```

Comment:
If the op is dropped before it is submitted to the ring, no work on the ring is scheduled.